package Broker;

//protoc -I=$SRC_DIR --cpp_out=$DST_DIR $SRC_DIR/addressbook.proto

enum DataType {
    INT32 = 0;
    DOUBLE = 1;
    BOOLEAN = 2;
    STRING = 3;
}

message DataDescriptor {
    optional string data_description = 1;
    required string data_name = 2;
    required string unit = 3;
    required DataType data_type = 4  [default = INT32];
}

message DataRegistration {
    optional string provider_name = 1;
    repeated DataDescriptor data_decriptors = 2;
}

message Data {
    required string data_name = 1;
    required DataType data_type = 2  [default = INT32];
    optional int32 data_int32 = 3;
    optional double data_double = 4;
    optional bool data_bool = 5;
    optional string data_string = 6;
}

message DataCollection {
    optional string provider_name = 1;
    repeated Data data_provided = 2;
}
